### ISSUE TITLE
Provide an elementary SDK

### ISSUE THREAD

**Issue by bilelmoussaoui** [flathub/flathub#421]
There's a lot of Elementary apps packaged as Flatpak packages.  I was wondering if it's wise to provide an elementary SDK which is basically GNOME SDK + stuff that is always used to build Vala apps in elementary: libgee, granite, elementary Gtk theme. I can provide a PR for this and maintain it.

This will allow third-party devs to package their apps as flatpak apps pretty easily

**Comment by nedrichards** [comment]
Not sure an SDK is the best option here. Both the ongoing security requirement and it's just a bit heavyweight. Perhaps a baseapp?

**Comment by nedrichards** [comment]
Also, is this the kind of thing that upstream elementary might be interested in? I know that they have their own well developed app store and developer experience but they may find some things useful.

**Comment by bilelmoussaoui** [comment]
Yeah, a baseapp could do the trick! @cassidyjames would you guys be interesting on something like that? I can maintain it and keep it updated.

**Comment by cassidyjames** [comment]
Yeah! I know some members of the team have experimented with something similar for Snaps, so I think we'd be open to it for Flatpak as well. With Snaps it's a "platform snap". I'm not sure what the correct thing is for Flatpak, but I understand it would be similar to how you have the GNOME 3.28 dependency, but this would be based on the FreeDesktop one with some additions. I don't think the entire contents of GNOME makes as much sense here, as elementary is based on FreeDesktop with some GNOME components, but not really a full downstream of GNOME.

**Comment by nedrichards** [comment]
@cassidyjames yeah - a 'platform snap' is closer to a runtime and is quite a lot of work (and requires rebuilds to bring in security updates in the base FDo - or any other dependencies, which is a non trivial ongoing burden). For this - you'd use the FDo runtime but then add a 'base app' alongside that of extra dependencies which would solely be the elementary stuff - we do the same thing for Electron apps as well and it works well. And because they'd be the same in every app they'd automatically get deuplicated by ostree as well.

@bilelmoussaoui I think that seems worth going ahead with.

**Comment by cassidyjames** [comment]
@codygarver might be able to provide guidance for which components (and versions) would be included.

**Comment by bilelmoussaoui** [comment]
Created the first step here https://github.com/flathub/flathub/pull/437
